docs/api/wanchain.md

```diff
@@ -1,5 +1,255 @@
 # API / Wanchain
 
+## Deploy gateway
+
+Deploy gateway allows for delegating deployment of an `AssetLedger` smart contract to a third party and transfering value in the same atomic transaction.
+
+To perform a delegate deploy through `DeployGateway`, you need to follow its flow:
+
+1. Maker (address creating the delegate deploy) defines the deploy (defines the data for the `AssetLedger` that will be created, who will be the owner of that `AssetLedger` and what amount of tokens will be sent to what address in exchange).
+2. Maker generates the deploy claim and signs it (`claim` functions).
+3. Maker approves value transfer.
+4. Maker sends deploy and signature to the Taker trough arbitrary channel.
+6. Taker performs the deploy.
+
+`Deploy` class is responsible for defining what will happen in the atomic swap. We support two different deploy configurations which we will call a fixed deploy and a dynamic deploy.
```

Comment:
`Deploy` class is responsible for defining what will happen in the atomic swap. We support two different deploy configurations, namely fixed deploy and dynamic deploy.